homeassistant/helpers/entity.py

```diff
@@ -723,11 +724,11 @@ def suggested_object_id(self) -> str | None:
             is type.__getattribute__(Entity, "name")
             # The check for self.platform guards against integrations not using an
             # EntityComponent and can be removed in HA Core 2024.1
-            and self.platform
+            and self.platform_data
```

Comment:
The comment about the deprecation and so on needs to be adjusted, here and in the other cases where we now check for `platform_data` instead of `platform`